### PR TITLE
NativeEngine: Override updateRenderTargetTextureSampleCount to prevent crashes

### PR DESCRIPTION
**Pull request by PolygonalSun**
A bug was found where Babylon Native would crash when opening files with certain extensions like Transmission.  The issue was that functions were being called within the ThinEngine that required the presence of a WebGL2RenderingContext (which the NativeEngine has no context for).  The code in this PR serves as a workaround so that this object be will avoided.  This allows Babylon Native to open previously unopenable files.  It should be noted that these file will render accurately when IBL is active.  When IBL is not active, the transmission extension will not render properly.  This issue will be addressed in a later PR.

Bug: https://github.com/BabylonJS/BabylonNative/issues/1166

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/13697/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13697/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/13697/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/13697/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/13697/merge
https://gui.babylonjs.com/?snapshot=refs/pull/13697/merge
https://nme.babylonjs.com/?snapshot=refs/pull/13697/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/13697/merge#BCU1XR#0

File: packages/dev/core/src/Engines/Extensions/engine.renderTarget.ts [hunk 1]
```diff
@@ -226,14 +226,14 @@ ThinEngine.prototype.updateRenderTargetTextureSampleCount = function (rtWrapper:
         return 1;
     }
 
+    samples = Math.min(samples, this.getCaps().maxMSAASamples);
```
Comment (bghgary) [inline packages/dev/core/src/Engines/Extensions/engine.renderTarget.ts:229]: I would separate this into another PR. These two places are not the only place that does this. We need to make sure they are consistent across the entire code base.

Comment (PolygonalSun) [reply packages/dev/core/src/Engines/Extensions/engine.renderTarget.ts:229]: Makes sense.  I'll rollback the changes to this line (and all related instances) and handle them in a separate PR.

**Review by sebavan — APPROVED** [review]
